src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -1704,6 +1704,43 @@ default Number sum() {
      */
     <U> Traversable<U> zipWithIndex(BiFunction<? super T, ? super Integer, ? extends U> mapper);
 
+    /**
+     * Converts this collection to a {@link SortedMap}.
+     *
+     * @param comparator  A comparator that induces an order of the Map keys.
+     * @param keyMapper   A function that maps an element to a key
+     * @param valueMapper A function that maps an element to a value
+     * @param merge A function that merges values that are associated with the same key
+     * @param <K>         The key type
+     * @param <V>         The value type
+     * @return A new {@link TreeMap}.
+     */
+    default <K, V> SortedMap<K, V> toSortedMap(
+            Comparator<? super K> comparator,
+            Function<? super T, ? extends K> keyMapper,
+            Function<? super T, ? extends V> valueMapper,
+            BiFunction<? super V, ? super V, ? extends V> merge) {
+        Objects.requireNonNull(comparator, "comparator is null");
+        return TreeMap.ofEntries(comparator, toMap(keyMapper, valueMapper, merge));
```

Comment:
Please add a shortcut for the empty case:

```java
if (isEmpty()) {
    return TreeMap.empty(comparator);
} else {
    ...
}
```

We could do the same for the other PRs, `toMap` and `toLinkedMap`.

Reply:
Please rename the `comparator` to `keyComparator` to clarify it a bit.

Reply:
I just did:
```Java
if (...) {
 ...
}
return ...;
```
as there's no real need for the `else` branch, let me know if you prefer the explicit style.
I also added the same case for `toMap`.